### DISCUSSION TITLE
Custom Template Parameter to register Simple Function

### DISCUSSION THREAD

**Discussion by tiagokepe** [facebookincubator/velox#3901] [Q&A] [answered]
Hi all and @Yuhta,

I've been implementing the new version for the function greatest/least, see issue #3728.

I would like to register the new functions using the STL Compare class as a template parameter, like:

```
Greatest: ExtremeValueFunction<std::greater>
Least: ExtremeValueFunction<std::less>

```
This standard is used in some STL containers and structures like [std::priority_queue](https://cplusplus.com/reference/queue/priority_queue/#:~:text=class%20Compare%20%3D%20less%3Ctypename%20Container%3A%3Avalue_type%3E)

But I didn't find any easy way to do that, I saw that Simple Function structures have already a top-level template parameter for specific purposes.

I would appreciate if you can give me any hint.

**Comment by pedroerp** [comment, marked as answer]
Looks like you want to do something similar what we do with trim:

> https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/StringFunctions.h#L359

(check the specializations below)

**Reply by tiagokepe** [reply]
Thanks @pedroerp , It looks like what I needed.
